src/Pagination.ts

```diff
@@ -0,0 +1,104 @@
+import { ItemProperties } from './interfaces';
+import { v, w } from '@dojo/widget-core/d';
+import { DNode } from '@dojo/widget-core/interfaces';
+import { theme, ThemeableMixin, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+import WidgetBase from '@dojo/widget-core/WidgetBase';
+import PageLink, { PageLinkProperties } from './pagination/PageLink';
+
+import * as css from './styles/pagination.m.css';
+
+export const PaginationBase = ThemeableMixin(WidgetBase);
+
+export interface PaginationProperties extends ThemeableProperties {
+	page: number;
+	pages: number;
+	statusMessage: string;
+	onPageRequest?: (pageNumber: number) => void;
```

Comment:
Can rewrite `onPageRequest?(pageNumber: number): void;` which is just a little easier to read.